Build failure: spotifyd

### Steps To Reproduce

Steps to reproduce the behavior:
1. build `spotifyd` from `nixos-unstable`
```
$ nix build -vL github:NixOS/nixpkgs/0ad13a6833440b8e238947e47bea7f11071dc2b2#spotifyd
```

### Build log

```
@nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/cjpyglnzfd5xnznf9dgmkvc2qjwd34z3-source
source root is source
Executing cargoSetupPostUnpackHook
unpacking source archive /nix/store/1lcybfnxks5n4dkw78rr6x5i8ab2zmfx-spotifyd-0.3.5-vendor.tar.gz
Finished cargoSetupPostUnpackHook
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
Executing cargoSetupPostPatchHook
Validating consistency between /build/source/Cargo.lock and /build/spotifyd-0.3.5-vendor.tar.gz/Cargo.lock
Finished cargoSetupPostPatchHook
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Running phase: buildPhase
Executing cargoBuildHook
++ env CC_X86_64_UNKNOWN_LINUX_GNU=/nix/store/qhpw32pz39y6i30b3vrbw5fw6zv5549f-gcc-wrapper-13.2.0/bin/cc CXX_X86_64_UNKNOWN_LINUX_GNU=/nix/store/qhpw32pz39y6i30b3vrbw5fw6zv5549f-gcc-wrapper-13.2.0/bin/c++ CARGO_TARGET_X86_64_UNKNOWN_LINUX_GNU_LINKER=/nix/store/qhpw32pz39y6i30b3vrbw5fw6zv5549f-gcc-wrapper-13.2.0/bin/cc CC_X86_64_UNKNOWN_LINUX_GNU=/nix/store/qhpw32pz39y6i30b3vrbw5fw6zv5549f-gcc-wrapper-13.2.0/bin/cc CXX_X86_64_UNKNOWN_LINUX_GNU=/nix/store/qhpw32pz39y6i30b3vrbw5fw6zv5549f-gcc-wrapper-13.2.0/bin/c++ CARGO_TARGET_X86_64_UNKNOWN_LINUX_GNU_LINKER=/nix/store/qhpw32pz39y6i30b3vrbw5fw6zv5549f-gcc-wrapper-13.2.0/bin/cc CARGO_BUILD_TARGET=x86_64-unknown-linux-gnu HOST_CC=/nix/store/qhpw32pz39y6i30b3vrbw5fw6zv5549f-gcc-wrapper-13.2.0/bin/cc HOST_CXX=/nix/store/qhpw32pz39y6i30b3vrbw5fw6zv5549f-gcc-wrapper-13.2.0/bin/c++ cargo build -j 24 --target x86_64-unknown-linux-gnu --frozen --profile release --no-default-features --features=alsa_backend,pulseaudio_backend,dbus_mpris
   Compiling libc v0.2.140
   Compiling autocfg v1.1.0
   Compiling proc-macro2 v1.0.54
   Compiling unicode-ident v1.0.8
   Compiling quote v1.0.26
   Compiling cfg-if v1.0.0
   Compiling version_check v0.9.4
   Compiling syn v1.0.109
   Compiling cc v1.0.79
   Compiling libm v0.2.6
   Compiling serde v1.0.159
   Compiling typenum v1.16.0
   Compiling serde_derive v1.0.159
   Compiling memchr v2.5.0
   Compiling once_cell v1.17.1
   Compiling log v0.4.17
   Compiling hashbrown v0.12.3
   Compiling pin-project-lite v0.2.9
   Compiling itoa v1.0.6
   Compiling pkg-config v0.3.26
   Compiling futures-core v0.3.27
   Compiling bitflags v1.3.2
   Compiling adler v1.0.2
   Compiling gimli v0.27.2
   Compiling futures-task v0.3.27
   Compiling futures-sink v0.3.27
   Compiling bytes v1.4.0
   Compiling futures-channel v0.3.27
   Compiling miniz_oxide v0.6.2
   Compiling rustc-demangle v0.1.22
   Compiling fixedbitset v0.4.2
   Compiling parking_lot_core v0.9.7
   Compiling futures-util v0.3.27
   Compiling smallvec v1.10.0
   Compiling futures-io v0.3.27
   Compiling tinyvec_macros v0.1.1
   Compiling pin-utils v0.1.0
   Compiling generic-array v0.14.7
   Compiling protobuf v2.28.0
   Compiling num-traits v0.2.15
   Compiling indexmap v1.9.3
   Compiling num-integer v0.1.45
   Compiling slab v0.4.8
   Compiling lock_api v0.4.9
   Compiling scopeguard v1.1.0
   Compiling tokio v1.27.0
   Compiling tinyvec v1.6.0
   Compiling tracing-core v0.1.30
   Compiling percent-encoding v2.2.0
   Compiling cpufeatures v0.2.6
   Compiling serde_json v1.0.95
   Compiling unicode-bidi v0.3.13
   Compiling object v0.30.3
   Compiling ryu v1.0.13
   Compiling semver v1.0.17
   Compiling fnv v1.0.7
   Compiling tracing v0.1.37
   Compiling rustc-serialize v0.3.24
   Compiling iana-time-zone v0.1.54
   Compiling syn v2.0.11
   Compiling petgraph v0.6.3
   Compiling thiserror v1.0.40
   Compiling http v0.2.9
   Compiling form_urlencoded v1.1.0
   Compiling ppv-lite86 v0.2.17
error[E0310]: the parameter type `T` may not live long enough
    --> /build/spotifyd-0.3.5-vendor.tar.gz/rustc-serialize/src/serialize.rs:1155:5
     |
1155 |     fn decode<D: Decoder>(d: &mut D) -> Result<Cow<'static, T>, D::Error> {
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     |     |
     |     the parameter type `T` must be valid for the static lifetime...
     |     ...so that the type `T` will meet its required lifetime bounds...
     |
note: ...that is required by this bound
    --> /build/rustc-1.76.0-src/library/alloc/src/borrow.rs:180:30
help: consider adding an explicit lifetime bound
     |
1151 | impl<'a, T: ?Sized + 'static> Decodable for Cow<'a, T>
     |                    +++++++++

   Compiling backtrace v0.3.67
   Compiling byteorder v1.4.3
   Compiling base64 v0.13.1
   Compiling unicode-normalization v0.1.22
   Compiling ring v0.16.20
   Compiling opaque-debug v0.3.0
   Compiling httparse v1.8.0
   Compiling thread-id v4.0.0
   Compiling getrandom v0.2.8
   Compiling socket2 v0.4.9
For more information about this error, try `rustc --explain E0310`.
error: could not compile `rustc-serialize` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
```

### Additional context

Looks like a problem with building `rustc-serialize`. May be related to #295198 .

### Notify maintainers

@marsam
@h7x4

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.7.9, NixOS, 24.05 (Uakari), 24.05.20240309.3030f18`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.20.5`
 - channels(root): `"nixos"`
 - channels(atif): `""`
 - nixpkgs: `/nix/store/nra828scc8qs92b9pxra5csqzffb6hpl-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/audio/spotifyd/default.nix
+++ b/pkgs/applications/audio/spotifyd/default.nix
@@ -9,16 +9,16 @@
 
 rustPackages.rustPlatform.buildRustPackage rec {
   pname = "spotifyd";
-  version = "0.3.5";
+  version = "0.3.5-unstable-2024-02-18";
 
   src = fetchFromGitHub {
     owner = "Spotifyd";
     repo = "spotifyd";
-    rev = "v${version}";
-    hash = "sha256-+P85FWJIsfAv8/DnQFxfoWvNY8NpbZ2xUidfwN8tiA8=";
+    rev = "ff2f7a06e54bf05afd57a0243dc9f67abc15f040";
+    hash = "sha256-nebAd4a+ht+blRP52OF830/Dm15ZPwRL4IPWmmT9ViM=";
   };
 
-  cargoHash = "sha256-j+2yEtn3D+vNRcY4+NnqSX4xRQIE5Sq7bentxTh6kMI=";
+  cargoHash = "sha256-6BRIMTrWTwvX3yIGEYEvigMT+n4EtaruMdrej2Dd49w=";
 
   nativeBuildInputs = [ pkg-config ];
 
@@ -40,7 +40,7 @@ rustPackages.rustPlatform.buildRustPackage rec {
   meta = with lib; {
     description = "An open source Spotify client running as a UNIX daemon";
     homepage = "https://spotifyd.rs/";
-    changelog = "https://github.com/Spotifyd/spotifyd/raw/v${version}/CHANGELOG.md";
+    changelog = "https://github.com/Spotifyd/spotifyd/blob/${src.rev}/CHANGELOG.md";
     license = licenses.gpl3Plus;
     maintainers = with maintainers; [ anderslundstedt Br1ght0ne marsam ];
     platforms = platforms.unix;

